src/bin/julialauncher.rs

```diff
@@ -292,57 +299,98 @@ fn run_app() -> Result<i32> {
         }
     }
 
-    // We set a Ctrl-C handler here that just doesn't do anything, as we want the Julia child
-    // process to handle things.
-    ctrlc::set_handler(|| ()).with_context(|| "Failed to set the Ctrl-C handler.")?;
+    // On *nix platforms we replace the current process with the Julia one.
+    // This simplifies use in e.g. debuggers, but requires that we fork off
+    // a subprocess to do the selfupdate and versiondb update.
+    #[cfg(not(windows))]
+    match unsafe { fork() } {
+        // NOTE: It is unsafe to perform async-signal-unsafe operations from
+        // forked multithreaded programs, so for complex functionality like
+        // selfupdate to work julialauncher needs to remain single-threaded.
+        // Ref: https://docs.rs/nix/latest/nix/unistd/fn.fork.html#safety
+        Ok(ForkResult::Parent { child, .. }) => {
+            // wait for the daemon-spawning child to finish
+            match waitpid(child, None) {
```

Comment:
Is there a reason to wait for the child process to finish here?

Reply:
If you don't wait for your child you get zombies.